docs/howto/use-with-nfs.md

```diff
@@ -0,0 +1,292 @@
+
+1. **Install Packages:**
+   On the NFS server, run:
+   ```bash
+   sudo apt install -y nfs-kernel-server nfs-common rpcbind krb5-user krb5-admin-server krb5-kdc
+   ```
+
+2. **Handle Kerberos Configuration Prompts:**
+   During the installation of `krb5-user`, you will be prompted to provide configuration details for Kerberos. Here's what to enter:
+
+   - **Default Kerberos version 5 realm:**
+     Enter the Kerberos realm name, which is the uppercase version of your domain. For example:
+     ```
+     EXAMPLE.COM
+     ```
+
+   - **Kerberos servers for your realm:**
+     Enter the hostname of the Key Distribution Center (KDC). Assuming the KDC is on the same host as the NFS server:
+     ```
+     server.example.com
+     ```
+
+   - **Administrative server for your Kerberos realm:**
+     Enter the hostname of the Kerberos admin server, which is also the same as the NFS server in this case:
+     ```
+     server.example.com
+     ```
+
+---
+
+#### Step 2: Configure Kerberos
+
+1. **Create the Realm:**
+   ```bash
+   sudo krb5_newrealm
+   ```
+   Follow the prompts to set up the Kerberos realm.
+
+2. **Add Principals:**
+   In Kerberos, a principal is a unique identity used for authentication.
+
+   - Add a principal for the NFS server:
+     This principal is used by the NFS client to authenticate when mounting an NFS directory.
+     ```bash
+     sudo kadmin.local addprinc -randkey nfs/server.example.com
+     ```
+
+   - Add a principal for the user `alice`:
+     This principal is used for authentication when the user accesses the mounted NFS directory.
+     ```bash
+     sudo kadmin.local addprinc alice
+     ```
+     Set a password for the user `alice`.
+
+3. **Generate Keytabs:**
+
+   A *keytab* is a file that contains Kerberos principals and their associated secret keys. It allows services (such as NFS) to authenticate without needing to input a password each time.
+
+   - Export the keytab for the NFS server and the user `alice`:
+     ```bash
+     sudo kadmin.local ktadd -k /etc/krb5.keytab nfs/server.example.com
+     ```
+
+---
+
+#### Step 3: Configure NFS Server
+
+1. **Create and Configure the Shared Directory:**
+
+   You’ll need to create the directory to share via NFS and configure the `/etc/exports` file to define it.
+
+   - **Create a directory owned by `alice`:**
+     ```bash
+     sudo mkdir -p /srv/nfs/shared/alice
+     sudo chown alice:alice /srv/nfs/shared/alice
+     ```
+
+   - **Configure exports:**
+     Edit the `/etc/exports` file to define the shared directory:
+     ```bash
+     sudo editor /etc/exports
+     ```
+     Add a line like this:
+     ```
+     /srv/nfs/shared *(rw,sync,no_subtree_check,sec=krb5)
+     ```
+
+2. **Configure IDMAP:**
+   Edit the IDMAP configuration:
+   ```bash
+   sudo editor /etc/idmapd.conf
+   ```
+   Ensure the following is set:
+   ```
+   [General]
+   Domain = example.com
+   ```
+
+3. **Restart Services:**
+   Restart the required services:
```

Comment:
Simple repeats the header.

If additional context can be given in the text then provide it, otherwise the text might be removed as redundant.

Reply:
> otherwise the text might be removed as redundant.

done